src/umpire/util/shared_memory_helper.hpp

```diff
@@ -0,0 +1,39 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-25, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_shared_memory_helper_HPP
+#define UMPIRE_shared_memory_helper_HPP
+
+#include <string>
+
+namespace umpire {
+namespace util {
+
+inline bool matchesSharedMemoryResource(const std::string& name, const std::string& resource_type) noexcept
+{
+  const std::string prefix = "SHARED::" + resource_type;
+
+  // Check if name starts with "SHARED::" + resource_type
+  if (name.find(prefix) == 0) {
+    return true;
+  }
+
+  // Check if name starts with "SHARED::" or "SHARED" AND that this resource_type is the default
+  if ((name.find("SHARED::") == 0) || (name == "SHARED")) {
+#ifdef UMPIRE_DEFAULT_SHARED_MEMORY_RESOURCE
+    return std::string(UMPIRE_DEFAULT_SHARED_MEMORY_RESOURCE) == resource_type;
```

Comment:
Would it make sense to use the variable `umpire::default_shared_memory_resource` defined in config.hpp?